### PR TITLE
Clean up code

### PR DESCRIPTION
**Pull request by NullHypothesis**
This PR combines many commits that clean up the code one way or another.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -74,10 +66,5 @@ func main() {
 		l.Fatal("We were told to use the 'any' interface but it's not present.")
 	}
 
-	go func() {
-		if err := http.ListenAndServe(":80", http.HandlerFunc(redirectToTLS)); err != nil {
-			l.Printf("ListenAndServe port 80 error: %v", err)
-		}
-	}()
-	l.Println(http.ListenAndServeTLS(":443", fullChain, privKey, nil))
+	l.Println(http.ListenAndServe(":8443", nil))
```
Comment (reethikar) [inline main.go]: In the case that we're exposing this over HTTP, we will have to test that the zerotrace code that extracts the connection works correctly with the reverse-proxy set up.., [as it extracts the *tls.Conn](https://github.com/brave-experiments/proxy-detection-server/blob/8e460d16ffa70a29662df7ccf6706aa807737180/zerotrace.go#L140)

(and zerotrace connection is made from the clientside using a [secure websocket](https://github.com/brave-experiments/proxy-detection-server/blob/8e460d16ffa70a29662df7ccf6706aa807737180/html_templates.go#L220))

File: handlers.go [hunk 2]
```diff
@@ -37,12 +39,20 @@ func measureHandler(w http.ResponseWriter, r *http.Request) {
...
 
-// pingHandler for ICMP measurements which also serves the webpage via a template
+// pingHandler implements a one-off measurement for the connecting client.  It
+// does the following:
+//
+//   1. Send ICMP packets to the client to determine the RTT.
+//   2. Serve the client JavaScript that initiates a WebSocket connection with
+//      us, again to determine the application-level RTT.
+//   3. Start another WebSocket connection to run a 0trace measurement, to
+//      determine an even more accurate RTT.
```
Comment (reethikar) [inline handlers.go:57]: Thanks for the documentation improvements, these are extremely clear

File: networking.go [hunk 3]
```diff
@@ -23,28 +23,15 @@ type FormDetails struct {
...
-
 type Results struct {
-	UUID       string
-	IPaddr     string
-	Timestamp  string
-	IcmpPing   PingMsmt
-	MinIcmpRtt float64
+	UUID      string
+	Timestamp string
+	PingStats *ping.Statistics
```
Comment (reethikar) [inline networking.go]: The [webpage results makes use of MinIcmpRtt](https://github.com/brave-experiments/proxy-detection-server/blob/8e460d16ffa70a29662df7ccf6706aa807737180/html_templates.go#L246) and needs to be changed too.

File: html_templates.go [hunk 4]
```diff
@@ -240,7 +240,7 @@ const pingPage = `
 
      getLatencyWebSocket().then((latencies) => {
         document.getElementById('uuid').innerHTML = {{ .UUID }};
-        document.getElementById('ip').innerHTML = {{ .IPaddr }};
+        document.getElementById('ip').innerHTML = {{ .PingStats.Addr }};
         document.getElementById('values').innerHTML = latencies;
         document.getElementById('data').innerHTML = min(latencies);
         document.getElementById('icmp').innerHTML = {{ .MinIcmpRtt }};
```
Comment (reethikar) [inline html_templates.go]: This `{{ .MinIcmpRtt }}` was part of the Results struct and could now be extracted from `.PingStats`.

**Review by reethikar — COMMENTED** [review]
Thanks for the PR, and the comments for the commit are very useful as well. Other than fixing the ping page HTML, everything else looks good.

File: main.go (hunk 1 above)
Comment (NullHypothesis) [reply main.go]: Noted! I've done some refactoring already and this doesn't look like a problem.

File: html_templates.go (hunk 4 above)
Comment (NullHypothesis) [reply html_templates.go]: Fixed in f9039f2.

**Review by reethikar — APPROVED** [review]
LGTM!

**Pull request merged**